exercises/forth/test/Tests.hs

```diff
@@ -1,74 +1,78 @@
 {-# LANGUAGE OverloadedStrings #-}
-import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts(..))
-import System.Exit (ExitCode(..), exitWith)
-import Forth (ForthError(..), evalText, empty, formatStack)
-import Control.Monad (foldM)
-import Data.Text (Text)
 
-exitProperly :: IO Counts -> IO ()
-exitProperly m = do
-  counts <- m
-  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess
+import Control.Monad     (foldM)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
 
-testCase :: String -> Assertion -> Test
-testCase label assertion = TestLabel label (TestCase assertion)
+import Forth (ForthError(..), empty, evalText, formatStack)
 
 main :: IO ()
-main = exitProperly $ runTestTT $ TestList
-       [ TestList forthTests ]
-
-runTexts :: [Text] -> Either ForthError Text
-runTexts xs = formatStack <$> foldM (flip evalText) empty xs
-
-forthTests :: [Test]
-forthTests =
-  [ testCase "no input, no stack" $
-    "" @=? formatStack empty
-  , testCase "numbers just get pushed onto the stack" $
-    Right "1 2 3 4 5" @=? runTexts ["1 2 3 4 5"]
-  , testCase "non-word characters are separators" $
-    -- Note the Ogham Space Mark ( ), this is a spacing character.
```

Comment:
I think that, ideally, people should get all the information needed to solve an exercise in the following order:

- Running the tests - As a general rule this should be enough to know what went wrong.
- Reading the hints - This could help when people get stuck.
- Inspecting the test suite - Forcing people to read code should be a last resort solution, but I understand that some things are too complex to explain in English.

> If there is someone who does not read the README but only look at the tests, the lack of this comment might make things confusing.

It could happen but, if `HINTS.md` is the right place to have the hints, we should enforce that there are no hints in the test suites or in the stub solutions.

IMHO, the only comments that should be in the test suites are the ones related to the test's implementation, to assist us in maintaining the tests suites.

An alternative would be to improve the tests and descriptions, if you think they are not good enough.

But this is just my opinion. If you think it is important to keep the comment there, I can update the PR immediately.